src/logger/logger.py

```diff
@@ -21,470 +24,521 @@
+    def inject_train_prio_fault(self, event: Event) -> Type[None]:
         """
         This function should be called when injecting a train prio fault into the simulation.
         This should include the fault configuration, the affected element, the value before and the
         value after the fault.
-        :param tick: The current simulation tick
-        :param train_prio_fault_configuration: The configuration of the fault
-        :param affected_element: The affected element
-        :param value_before: The value before the fault
-        :param value_after: The value after the fault
-        :rtype: None
+        :param event: the event containing all relevant info
         """
+        if "train_prio_fault_configuration" not in event.arguments:
+            return
         InjectFaultLogEntry.create(
             timestamp=datetime.now(),
-            tick=tick,
-            message=f"Train prio fault with configuration {train_prio_fault_configuration} "
-            f"on element {affected_element} changed from {value_before} to {value_after}",
+            tick=event.arguments["tick"],
+            message=f"Train prio fault with configuration "
+            f"{event.arguments['train_prio_fault_configuration']} "
+            f"on element {event.arguments['affected_element']} changed from "
+            f"{event.arguments['value_before']} to {event.arguments['value_after']}",
             run_id=self.run_id,
-            train_prio_fault_configuration=train_prio_fault_configuration,
-            affected_element=affected_element,
-            value_before=value_before,
-            value_after=value_after,
+            train_prio_fault_configuration=event.arguments[
+                "train_prio_fault_configuration"
+            ],
+            affected_element=event.arguments["affected_element"],
+            value_before=event.arguments["value_before"],
+            value_after=event.arguments["value_after"],
         )
 
-    def inject_train_speed_fault(
-        self,
-        tick: int,
-        train_speed_fault_configuration: UUID,
-        affected_element: str,
-        value_before: str,
-        value_after: str,
-    ) -> Type[None]:
+    def inject_train_speed_fault(self, event: Event) -> Type[None]:
         """
         This function should be called when injecting a train speed fault into the simulation.
         This should include the fault configuration, the affected element, the value before and the
         value after the fault.
-        :param tick: The current simulation tick
-        :param train_speed_fault_configuration: The configuration of the fault
-        :param affected_element: The affected element
-        :param value_before: The value before the fault
-        :param value_after: The value after the fault
-        :rtype: None
+        :param event: the event containing all relevant info
         """
+        if "train_speed_fault_configuration" not in event.arguments:
+            return
```

Comment:
Why did you add this? I think that should not ever happen right?

Reply:
This can be the case, because there are only the EventTypes `INJECT_FAULT` and `RESOLVE_FAULT`. This results in all of the specific faults have to check, if they are meant by the call or it addresses another fault type.